src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4882,8 +4882,6 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           }
           NoSymbol
         }
-        if (phase.erasedTypes && qual.isInstanceOf[Super] && tree.symbol != NoSymbol)
-          qual setType tree.symbol.owner.tpe
```

Comment:
This hack is almost 12 years old.. https://github.com/scala/scala/commit/2a5f62338c#diff-b0ec9c916463796fd7ff1d2ed73d46e2L810